### PR TITLE
BSIS-796 Donations Report - report ui-grid export POC

### PR DESCRIPTION
**Pull request by lauravignoli**

### CODE REVIEW

**Comment by timler** [comment]
Noting that the build failed due to the download of phantomJS

File: app/views/reports/donationsReport.html [hunk 1]
```diff
@@ -3,9 +3,37 @@
...
+                </small>
+            </div>
+        </div>
+      </div>
+
+      <div class="col-sm-offset-2 ">
+          <button class="btn btn-primary" type="submit" ng-disabled="searching">{{searching ? "Searching..." :
+              "Search"}}
+          </button>
+          <button class="btn btn-primary" type="reset" ng-click="clearSearch(selectPeriodForm)">Clear</button>
```
Comment (timler) [inline app/views/reports/donationsReport.html]: To fix the problem with clearing the form (that the "master" start/end dates aren't persisted, replace `type="reset"` with `type="button"`. What was happening is that the default reset functionality was clearing the values in the form fields (after the default was restored).

File: app/scripts/controllers/reports/donationsReport.js [hunk 2]
```diff
@@ -1,10 +1,77 @@
...
+
+    $scope.clearSearch = function(form) {
+      $scope.search = angular.copy(master);
+      form.$setPristine();
+      form.$setUntouched();
+      $scope.gridOptions.data = [];
+    };
+
+    $scope.getReport = function(selectPeriodForm) {
+      if (selectPeriodForm.$valid) {
```
Comment (timler) [inline app/scripts/controllers/reports/donationsReport.js]: An early return would make the code more readable - i.e. return if the form is not valid.

File: app/views/reports/donationsReport.html (hunk 1 above)
Comment (lauravignoli) [inline app/views/reports/donationsReport.html]: Fixed here 574d1d6b376ee97a5eda8b81455aecb00f2f179e

File: app/scripts/controllers/reports/aboRhGroupsReport.js [hunk 3]
```diff
@@ -138,13 +138,19 @@ angular.module('bsis')
       paginationPageSizes: [10],
       paginationTemplate: 'views/template/pagination.html',
       columnDefs: columnDefs,
+      
+      exporterPdfOrientation: 'landscape',
+      exporterPdfPageSize: 'A4',
+      exporterPdfDefaultStyle: {fontSize: 8, margin: [-2, 0, 0, 0] },
+      exporterPdfTableHeaderStyle: {fontSize: 8, bold: true, margin: [-2, 0, 0, 0] },
+      exporterPdfMaxGridWidth: 650,
```
Comment (timler) [inline app/scripts/controllers/reports/aboRhGroupsReport.js]: I believe it needs to be 550 in order to fit on the page landscape. I just printed out the PDF and the table is cut off on the right. I worked on the Test Batch report and noticed that the ui-grid documentation was incorrect.

File: app/views/reports/aboRhGroupsReport.html [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <div class="col-sm-3 col-md-2" ng-include="'views/reports/sidebar.html'"></div>
 <div class="col-sm-9 col-md-10 main">
-  <span class="h3 page-header">Donations report</span>
+  <span class="h3 page-header">Abo Rh Groups Report</span>
```
Comment (timler) [inline app/views/reports/aboRhGroupsReport.html]: It should be ABO not Abo - important for display purposes.

File: app/scripts/controllers/reports/aboRhGroupsReport.js [hunk 5]
```diff
@@ -23,6 +23,16 @@ angular.module('bsis')
...
+    function createEmptyRow() {
+
+      var emptyRow = {};
+      emptyRow.cohorts = [{}, {option:''}, {option:''}];
+      emptyRow.venue = {};
+      emptyRow.venue.name = '';
+      return emptyRow;
+
+    }
+
```
Comment (timler) [inline app/scripts/controllers/reports/aboRhGroupsReport.js]: If possible, we should avoid adding a blank row. What I'd suggest is extracting the logic from the forEach loop into a function and calling that function an extra time after the loop finishes. Or using a different loop construct (e.g. do while) where the code can be executed an extra time - after all the rows have been processed.

**Comment by timler** [comment]
Nice. The report is starting to look really great!

File: app/scripts/controllers/reports/aboRhGroupsReport.js (hunk 5 above)
Comment (lauravignoli) [inline app/scripts/controllers/reports/aboRhGroupsReport.js]: Fixed here 444462a691902260c6c621383d943f2086f29d7c

**Pull request merged**